### PR TITLE
Task/excc support

### PR DESCRIPTION
**Pull request by pingwindyktator**

### CODE REVIEW

**Review by siy — APPROVED** [review]

File: stratum/stratum.go [hunk 1]
```diff
@@ -932,7 +931,7 @@ func (s *Stratum) PrepWork() error {
 		return err
 	}
 
-	var workData [192]byte
+	var workData [work.GetworkDataLen]byte
 	copy(workData[:], workdata[:])
```
Comment (harzo) [inline stratum/stratum.go]: Is it necessary to creating new `workData` here? What about creating 320 long in a first place and just using it?

File: stratum/stratum.go [hunk 2]
```diff
@@ -951,10 +950,7 @@ func (s *Stratum) PrepWork() error {
 		return nil
 	}
 
-	w := work.NewWork(workData, s.Target, givenTs, uint32(time.Now().Unix()), false)
-
-	log.Tracef("Stratum prepated work data %v, target %032x",
-		hex.EncodeToString(w.Data[:]), w.Target.Bytes())
+	w := work.NewWork(bh, s.Target, givenTs, uint32(time.Now().Unix()), false)
```
Comment (harzo) [inline stratum/stratum.go:948]: Hmm. Now it seems that `workData` is not used at all, neither what is inside it. Only TimestampWord

File: stratum/stratum.go [hunk 3]
```diff
@@ -963,8 +959,7 @@ func (s *Stratum) PrepWork() error {
 // PrepSubmit formats a mining.sumbit message from the solved work.
 func (s *Stratum) PrepSubmit(data []byte) (Submit, error) {
 	log.Debugf("Stratum got valid work to submit %x", data)
-	log.Debugf("Stratum got valid work hash %v",
-		chainhash.HashH(data[0:180]))
+	log.Debugf("Stratum got valid work hash %v", chainhash.HashH(data[0:180]))
 	data2 := make([]byte, 180)
 	copy(data2, data[0:180])
```
Comment (harzo) [inline stratum/stratum.go]: `data2` is not used

File: getwork.go [hunk 4]
```diff
@@ -165,36 +166,34 @@ func GetWork() (*work.Work, error) {
...
-			len(target))
+		return nil, fmt.Errorf("Wrong target length: got %d, expected 32", len(target))
 	}
 
 	bigTarget := new(big.Int)
 	bigTarget.SetBytes(util.Reverse(target))
 
-	var workData [192]byte
+	var workData [work.GetworkDataLen]byte
 	copy(workData[:], data)
```
Comment (harzo) [inline getwork.go]: Pointless copying. We don't need `workData` at all.

**Review by harzo — COMMENTED** [review]
It would be perfect, since Go 1.11 is released, to update Go, remove Gopkg.json and Gopkg.lock, and use go build, go test again ;) Go 1.11 has build-in package manager - vgo
Just a suggestion :)

**Comment by harzo** [comment]
Looks good :+1:

**Review by harzo — APPROVED** [review]

**Pull request merged**